examples/seeder_example/vt_application.cpp

```diff
@@ -0,0 +1,1149 @@
+	{ Statistics::TaskController, tcStatistics_Container },
+	{ Statistics::Credits, credits_Container }
+};
+
+const std::array<std::uint16_t, SeederVtApplication::NUMBER_ONSCREEN_SECTIONS> SeederVtApplication::SECTION_STATUS_OUTRECTS = {
+	section1Status_OutRect,
+	section2Status_OutRect,
+	section3Status_OutRect,
+	section4Status_OutRect,
+	section5Status_OutRect,
+	section6Status_OutRect
+};
+
+const std::array<std::uint16_t, SeederVtApplication::NUMBER_ONSCREEN_SECTIONS> SeederVtApplication::SECTION_SWITCH_STATES = {
+	section1EnableState_ObjPtr,
+	section2EnableState_ObjPtr,
+	section3EnableState_ObjPtr,
+	section4EnableState_ObjPtr,
+	section5EnableState_ObjPtr,
+	section6EnableState_ObjPtr
+};
+
+const std::array<SeederVtApplication::ActiveScreen, static_cast<std::uint8_t>(SeederVtApplication::Alarm::NumberOfAlarms)> SeederVtApplication::ALARM_SCREENS = {
+	ActiveScreen::NoMachineSpeed,
+	ActiveScreen::NoTaskController
+};
+
+const std::array<std::uint16_t, static_cast<std::uint8_t>(SeederVtApplication::Alarm::NumberOfAlarms)> SeederVtApplication::ALARM_DESCRIPTION_LINES = {
+	NoMachineSpeed_OutStr,
+	NoTaskController_OutStr
+};
+
+SeederVtApplication::SeederVtApplication(std::shared_ptr<isobus::PartneredControlFunction> VTPartner, std::shared_ptr<isobus::PartneredControlFunction> TCPartner, std::shared_ptr<isobus::InternalControlFunction> source) :
+  TCClientInterface(TCPartner, source, nullptr),
+  VTClientInterface(VTPartner, source),
+  txFlags(static_cast<std::uint32_t>(UpdateVTStateFlags::NumberOfFlags), processFlags, this),
+  speedMessages(source, false, false, false, false)
+{
+}
+
+bool SeederVtApplication::initialize()
+{
+	bool retVal = true;
+
+	sectionControl.set_number_of_sections(NUMBER_ONSCREEN_SECTIONS);
+
+	objectPool = isobus::IOPFileInterface::read_iop_file("BasePool.iop");
+
+	if (objectPool.empty())
+	{
+		std::cout << "Failed to load object pool from BasePool.iop" << std::endl;
+		return false;
+	}
+	std::cout << "Loaded object pool from BasePool.iop" << std::endl;
+
+	// Generate a unique version string for this object pool (this is optional, and is entirely application specific behavior)
+	std::string objectPoolHash = isobus::IOPFileInterface::hash_object_pool_to_version(objectPool);
+
+	VTClientInterface.set_object_pool(0, isobus::VirtualTerminalClient::VTVersion::Version4, objectPool.data(), objectPool.size(), objectPoolHash);
+	softkeyEventListener = VTClientInterface.add_vt_soft_key_event_listener([this](const isobus::VirtualTerminalClient::VTKeyEvent &event) { this->handle_vt_key_events(event); });
+	buttonEventListener = VTClientInterface.add_vt_button_event_listener([this](const isobus::VirtualTerminalClient::VTKeyEvent &event) { this->handle_vt_key_events(event); });
+	numericValueEventListener = VTClientInterface.add_vt_change_numeric_value_event_listener([this](const isobus::VirtualTerminalClient::VTChangeNumericValueEvent &event) { this->handle_numeric_value_events(event); });
```

Comment:
I think you can remove the lambda functions here and just pass the functions directly to the `add_..._listener` functions

Reply:
Am I missing something on these? 
![image](https://github.com/Open-Agriculture/AgIsoStack-plus-plus/assets/10929341/a7097e56-a2d4-4ffe-bb52-ccf3ca354203)

Reply:
Huhh haha, are you able to compile the vt version3_object_pool example? We also use direct calls there: https://github.com/Open-Agriculture/AgIsoStack-plus-plus/blob/52033f9874c8313557a59bc7bda4c8e39dd04530/examples/virtual_terminal/version3_object_pool/main.cpp#L142-L143

Reply:
Maybe try linking `this->handle_vt_key_events` instead? It being a non-static function of the class seems the only difference to me

Reply:
Same results unfortunately. I think the one being static must make all the difference 😣
![image](https://github.com/Open-Agriculture/AgIsoStack-plus-plus/assets/10929341/c4f9a6df-369a-471e-99d5-5510e1d1a8c4)

The lambdas might have to stay unfortunately

Reply:
Ahh I see, `a pointer to a bound function may only be used to call the function` is the reason why that won't work. The lambda is fine, it would've just made it more readable but this lambda function still is perfectly fine with me :)